src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -423,6 +444,19 @@ def _offer_aggregator(self, arguments):
                                  f"on area {self.device.name} with arguments {arguments}.",
                 "transaction_id": arguments.get("transaction_id", None)}
 
+    def _update_bid_aggregator(self, arguments):
+        assert set(arguments.keys()) == {'price', 'energy', 'type', 'transaction_id'}
+        existing_bids = list(self.get_posted_bids(self.market))
+        for bid in existing_bids:
+            assert bid.buyer == self.owner.name
+            if bid.id in self.market.bids.keys():
+                bid = self.market.bids[bid.id]
+            self.market.delete_bid(bid.id)
+
+            self.remove_bid_from_pending(self.market.id, bid.id)
+        self.post_bid(self.market, arguments["price"], arguments["energy"],
+                      buyer_origin=bid.buyer_origin)
+
```

Comment:
This is basically a new feature. Right? The "normal" device client does not have one of those. When was this decided?

Reply:
This is after colin notice a bug where energy requirement was more than zero so client posted a bid/ offer. However since agrgegator is batch_processing that info in the same time old offer/bid got traded. SO, I propose him to actually add a new method called `update_offer/update` which would only check if the old offer/bid is not yet traded and if still persist then update the price

Reply:
I see, thanks for explaining. 
But again, this favours the aggregator before the single-device api. Me, as a single user must first list all bids, then delete the old one and then post a new one.  
IMO the case you are mentioning, is not a bug but something we just have to catch and report back to the aggregator. 
So, either we add it to the "normal" device too or we just say now: The aggregator as it is now will be thrown away anyway so need for that.

Reply:
That case is not visible in the device since the device client request is instantly processed (bid/delete_bid/delete_offer/offer) and updates the `energy_requirement/available_energy` where as in case of aggregator case it is being cached and then batch processing is done every tick, so there could be a case where old bid is in cache in the mean time client posted the new one and in between the old one got accepted. However, still I would be in favour supporting device client with update_offer/update_bid as it looks more cleaner to me than client deleting the old bid and posting bid with updated price